drivers/pinctrl/pinctrl_xmc4xxx.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_xmc4xxx_pinctrl
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/xmc4xxx-pinctrl.h>
+
+#include <xmc_gpio.h>
+
+static const struct device *const gpio_devs[] = {
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio0)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio1)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio2)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio3)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio4)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio5)),
+};
```

Comment:
this array is only used to check port_id size?

Reply:
It's used in the device_is_ready() above.